### PR TITLE
fix: refresh ds config on env switch

### LINKED ISSUE #26206: [Bug]: Generate page with data -  On switching from staging [not configured] to prod [configured] - get ds structure fails
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Being on Staging environment [for which DS is not configured], when I generate page with data, and then switch environment to Prod, though prod ds is configured, the Select Table call fails. 

### Steps To Reproduce

1. Login as a user with access to Multiple Environments. 
2. Configure any DS [say S3 or PostGreSQL] for Production environment
3. Don't configure Staging
4. Switch the bottom dropdown to Staging.
5. Click on Generate page with Data from the pages Add icon. 
6. Select the created DS
7. Note that there is an error in the Select Table/Bucket dropdown, as Staging is not configured.
8. Now switch the environment dropdown to Production
9. Note that the Select Table/Bucket dropdown refreshes, but the data is not retrieved and the same error as point 7 is seen though the DS is configured.

Uploading Screen Recording 2023-08-09 at 5.15.28 PM.mov…

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by ayushpahwa**
## Description
Generate Page updates based on env switch
- Whenever env is switched, the schema is refetched. Based on this, the datasource options for the dropdown have to be refreshed (since that causes the isValid flag to change)
- Update the selected ds option based on the update of ds options
- Hide searchable columns if the datasource schema is fetching or schema fetched fails

#### PR fixes following issue(s)
Fixes #26206 

#### Type of change

- Bug fix (non-breaking change which fixes an issue)

- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Issues raised during DP testing
https://github.com/appsmithorg/appsmith/pull/26667#issuecomment-1698704257
>
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [x] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by btsgh** [comment]
Need an EE DP to test this issue.

File: app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/GeneratePageForm.tsx [hunk 1]
```diff
@@ -401,15 +402,22 @@ function GeneratePageForm() {
     workspace?.userPermissions || [],
   );
 
+  const spreadSheetsProps = useSpreadSheets({
+    setSelectedDatasourceTableOptions,
+    setSelectedDatasourceIsInvalid,
+  });
+
+  const fetchingDatasourceConfigs =
```
Comment (ChandanBalajiBP) [inline app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/GeneratePageForm.tsx:410]: Can you please add comments here and above

**Review by ChandanBalajiBP — APPROVED** [review]
Looks good, but add comments in case you are doing another commit else add it from other PR's

File: app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/GeneratePageForm.tsx (hunk 1 above)
Comment (ayushpahwa) [reply app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/GeneratePageForm.tsx:411]: I just moved the function from bottom to top. I added the comments from what I know of the functions

**Review by ChandanBalajiBP — APPROVED** [review]

**Comment by ayushpahwa** [comment]
Adding test plan approved since this was moved to qa done by @btsgh

**Pull request merged**